packages/core/core/src/PackagerRunner.js

```diff
@@ -40,15 +40,20 @@ export default class PackagerRunner {
     let filePath = nullthrows(bundle.filePath);
     let dir = path.dirname(filePath);
     if (!this.distExists.has(dir)) {
-      await mkdirp(dir);
+      await fs.mkdirp(dir);
       this.distExists.add(dir);
     }
 
+    // Use the file mode from the entry asset as the file mode for the bundle
+    let options = {
+      mode: (await fs.stat(bundle.assetGraph.getEntryAssets()[0].filePath)).mode
```

Comment:
I don't think this should be done for browser targets. Having a file with the executable bit set might have unexpected security issues on the server.

Also, shouldn't this rather copy only the executable bit and leave read/write on default?

Reply:
@wbinnssmith ?

Reply:
> This is good for another review

...

Reply:
Good call. I had assumed you were referring to hashbangs before :)

I agree the executable bit isn't necessary and only creates concerns with webservers. Not sure about the other bits of permissions, but you could probably make the argument they do the same (why should the permissions of source code in a build environment reflect the permissions of the build target run on a webserver?)